tools/cellprofiler/display_data_on_image.xml

```diff
@@ -213,30 +215,36 @@ f.close()
             </param>
             <when value="Text">
                 <param name="text_color" type="color" label="Text color" />
-                <param name="font_size" type="text" label="Font size(points)" />
-                <param name="number_of_decimals" type="text" label="Number of decimals" />
-                <param name="annotation_offset" type="text" label="Annotation offset (in pixels)" />
+                <param name="font_size" type="integer" label="Font size(points)" value="11">
+                    <validator type="in_range" min="8" message="minimum size of font is 8" />
+                </param>
+                <param name="number_of_decimals" type="integer" label="Number of decimals" value="0"/>
+                <param name="annotation_offset" type="integer" label="Annotation offset (in pixels)" value="0" />
             </when>
             <when value="Color">
-                <param name="color_map" label="Color map" type="select">
+                <param name="color_map" label="Color map" type="select" help="DisplayDataOnImage assigns a color to each object’s measurement value from a colormap when in colormap-mode, mapping the value to a color along the colormap’s continuum. This mapping has implicit upper and lower bounds to its range which are the extremes of the colormap. This setting determines whether the extremes are the minimum and maximum ">
                     <option value="Default">Default</option>
                 </param>
                 <conditional name="con_color_map_scale">
-                    <param name="color_map_scale" label="Color map scale" type="select">
+                    <param name="color_map_scale" label="Color map scale" type="select" help="assigns a color to each object’s measurement value from a colormap when in colormap-mode, mapping the value to a color along the colormap’s continuum. This mapping has implicit upper and lower bounds to its range which are the extremes of the colormap. This setting determines whether the extremes are the minimum and maximum values of the measurement from among the objects in the current image or manually-entered extremes">
                         <option value="Use this image's measurement range">Use this image's measurement range</option>
                         <option value="Manual">Manual</option>
                         <sanitizer sanitize="false"/>
                     </param>        
                     <when value="Manual">
-                        <param name="min_color_map_range" label="Min. color map range" type="text" />
-                        <param name="max_color_map_range" label="Max. color map range" type="text" />
+                        <param name="min_color_map_range" label="Min. color map range" type="float" help="This setting determines the lower bounds of the values for the color map" value="0.0" >
+                            <validator type="in_range" min="0.0" />

```

Comment:
@sunyi000 for this you can simply write `min="0.0"` into the main "param". No explicit validator needed.

Reply:
`min=0.0`, `max=1.0`